src/ol/color.js

```diff
@@ -23,6 +20,109 @@ import {clamp} from './math.js';
+  /^rgba?\(\s*(\d+%?)\s+(\d+%?)\s+(\d+%?)(?:\s*\/\s*(\d+%|\d*\.\d+|[01]\.?))?\s*\)$/i;
+const rgbLegacyAbsoluteRegEx =
+  /^rgba?\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)(?:\s*,\s*(\d+%|\d*\.\d+|[01]\.?))?\s*\)$/i;
+const rgbLegacyPercentageRegEx =
+  /^rgba?\(\s*(\d+%)\s*,\s*(\d+%)\s*,\s*(\d+%)(?:\s*,\s*(\d+%|\d*\.\d+|[01]\.?))?\s*\)$/i;
+const hexRegEx = /^#([\da-f]{3,4}|[\da-f]{6}|[\da-f]{8})$/i;
+
+/**
+ * @param {string} s Color component as number or percentage.
+ * @param {number} divider Divider for percentage.
+ * @return {number} Color component.
+ */
+function toColorComponent(s, divider) {
+  return s.endsWith('%')
+    ? Number(s.substring(0, s.length - 1)) / divider
+    : Number(s);
+}
+
+/**
+ * @param {string} color Color string.
+ */
+function throwInvalidColor(color) {
+  throw new Error('failed to parse "' + color + '" as color');
+}
+
+/**
+ * @param {string} color Color string.
+ * @return {Color} RGBa color array.
+ */
+function parseRgba(color) {
+  // Fast lane for rgb(a) colors
+  const rgb =
+    color.match(rgbLegacyAbsoluteRegEx) ||
+    color.match(rgbModernRegEx) ||
+    color.match(rgbLegacyPercentageRegEx);
+  if (rgb) {
+    const alpha = rgb[4];
+    const rgbDivider = 100 / 255;
+    return [
+      clamp((toColorComponent(rgb[1], rgbDivider) + 0.5) | 0, 0, 255),
+      clamp((toColorComponent(rgb[2], rgbDivider) + 0.5) | 0, 0, 255),
+      clamp((toColorComponent(rgb[3], rgbDivider) + 0.5) | 0, 0, 255),
+      alpha !== undefined ? clamp(toColorComponent(alpha, 100), 0, 1) : 1,
+    ];
+  }
+  // Fast exit for inalid rgb(a) colors
+  if (color.toLocaleLowerCase().startsWith('rgb')) {
+    throwInvalidColor(color);
+  }
+  // Fast lane for hex colors (also with alpha)
+  if (hexRegEx.test(color)) {
+    const hex = color.substring(1);
+    const step = hex.length <= 4 ? 1 : 2;
+    const colorFromHex = [0, 0, 0, 255];
+    for (let i = 0, ii = hex.length; i < ii; i += step) {
+      let colorComponent = parseInt(hex.substring(i, i + step), 16);
+      if (step === 1) {
+        colorComponent += colorComponent << 4;
+      }
+      colorFromHex[i / step] = colorComponent;
+    }
+    colorFromHex[3] = colorFromHex[3] / 255;
+    return colorFromHex;
+  }
+  // Fast exit for invalid hex colors
+  if (color.startsWith('#')) {
+    throwInvalidColor(color);
+  }
+
+  const context = getColorParseContext();
+  context.clearRect(0, 0, 1, 1);
+  context.fillStyle = '#abcdef';
+  let invalidCheckFillStyle = context.fillStyle;
+  context.fillStyle = color;
+  if (context.fillStyle === invalidCheckFillStyle) {
+    context.fillStyle = '#fedcba';
+    invalidCheckFillStyle = context.fillStyle;
+    context.fillStyle = color;
+    if (context.fillStyle === invalidCheckFillStyle) {
+      throwInvalidColor(color);
+    }
+  }
+  context.fillRect(0, 0, 1, 1);
+  const colorFromImage = Array.from(context.getImageData(0, 0, 1, 1).data);
+  colorFromImage[3] = Math.round((colorFromImage[3] / 255) * 100) / 100;
```

Comment:
Alpha can have 3 decimal places in Chrome, e.g.

context.fillStyle = '#fffffffe';
console.log(context.fillStyle); // 'rgba(255, 255, 255, 0.996)'

Reply:
Ok, that will break existing tests, but happy to change it.